ccache.c

```diff
@@ -2395,8 +2395,8 @@ cc_process_args(struct args *args, struct args **preprocessor_args,
 			continue;
 		}
 
-		// Alternate form of -o with no space.
-		if (str_startswith(argv[i], "-o")) {
+		// Alternate form of -o with no space. Ignore nvcc -odir.
+		if (str_startswith(argv[i], "-o") && !str_eq(argv[i], "-odir")) {
```

Comment:
Fixed it and added a test for it. Actually, the option ```-optf``` caused the same problem, so I fixed that as well. Since nvcc can't handle the ```-o``` option without space, that part is not ignored if nvcc was detected.